### PR TITLE
student course join: ask to update profile if there are no open sessions #2891

### LINKED ISSUE #2891: student course join: ask to update profile if there are no open sessions
Sometimes instructors send out join emails when there are no open sessions. In that case, we should inform the student that 
1. there are no pending sessions, 
2. TEAMMATES will send a notification when a session opens
3. recommended to update the profile

### PR DESCRIPTION
**Pull request by kanghj**
Fixes #2891

### CODE REVIEW

**Comment by kanghj** [comment]
On joining the course, if the profile has more than one unfilled attribute out of shortName, profile picture, nationality, email and moreInfo, this is what the student sees:
novisiblesessions

If there is only one attribute unfilled:
existingstudent

If the student has completed the profile, he/she will not get an extra message to update the profile:
fullprofile

**Comment by damithc** [comment]
@thyageshm could you review the code?
@kanghj  no need to say 'evaluation/feedback session'. We can simply say 'feedback session'

**Comment by thyageshm** [comment]
@kanghj nice work! a suggestion for the second case described above:
"Meanwhile, you can provide a preferred name that you would prefer to be called by". Here the "preferred" word is redundant and somehow doesn't sound right. I would suggest removing it, what are your thoughts?

**Comment by kanghj** [comment]
Changes made. @thyageshm, thanks for the suggestion, I did not notice the extra "preferred" in the message.

File: src/test/java/teammates/test/cases/ui/browsertests/StudentCourseJoinConfirmationPageUiTest.java [hunk 1]
```diff
@@ -70,12 +71,17 @@ private void testJoinNewConfirmation() {
         studentHomePage = createCorrectLoginPageType(browser.driver.getPageSource())
                            .loginAsStudent(TestProperties.inst().TEST_STUDENT1_ACCOUNT,
                                                   TestProperties.inst().TEST_STUDENT1_PASSWORD);
-        studentHomePage.verifyStatus(
-                String.format(Const.StatusMessages.STUDENT_COURSE_JOIN_SUCCESSFUL, courseId));
+        
+        String expectedStatus = String.format(Const.StatusMessages.STUDENT_COURSE_JOIN_SUCCESSFUL, "[" + courseId + "] " + courseName) + '\n' 
+                                + String.format(Const.StatusMessages.HINT_FOR_NO_SESSIONS_STUDENT, "[" + courseId + "] " + courseName) + '\n'  
+                                + "Meanwhile, you can update your profile here.";
```
Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/StudentCourseJoinConfirmationPageUiTest.java]: Hey @kanghj, just a clarification, we expect to see Const.StatusMessages.STUDENT_UPDATE_PROFILE here right? That constant's value seems a little different from the value you are checking here. Does the test pass?

Comment (kanghj) [inline src/test/java/teammates/test/cases/ui/browsertests/StudentCourseJoinConfirmationPageUiTest.java]: Yes the test passes. The value is different as I did not include the <a></a> tags in the expected status message. AppPage.verifyStatus calls getText() on the statusMessage, which returns the innerText of the element. The html tags (<a href ...>) in the status message are not included in the innerText. 

The status message is tested again in StudentCourseJoinAuthenticatedActionTest, this time with the <a></a> tag included.

Comment (thyageshm) [inline src/test/java/teammates/test/cases/ui/browsertests/StudentCourseJoinConfirmationPageUiTest.java]: ok!

**Comment by thyageshm** [comment]
The changes seem to be alright :-) Is it dev green?

**Comment by kanghj** [comment]
Yup, just ran the tests again and everything passed (:

**Comment by thyageshm** [comment]
You can merge it then :-)

**Pull request merged**